spec/source/locales-currencies-tz.html

```diff
@@ -0,0 +1,163 @@
+        2. If the string length of _normalized_ is not 3, then
+          1. return `false`.
+        3. If _normalized_ contains any character that is not in the range *"A"* to *"Z"* (`U+0041` to `U+005A`), then
+          1. return `false`.
+        4. Return `true`.
+      </emu-alg>
+    </emu-clause>
+
+  </emu-clause>
+
+  <emu-clause id="sec-time-zone-names">
+    <h1>Time Zone Names</h1>
+
+    <p>
+      The ECMAScript 2015 Internationalization API Specification identifies time zones using the Zone and Link names of the IANA Time Zone Database. Their canonical form is the corresponding Zone name in the casing used in the IANA Time Zone Database.
+    </p>
+
+    <p>
+      All registered Zone and Link names are allowed. Implementations must recognize all such names, and use best available current and historical information about their offsets from UTC and their daylight saving time rules in calculations. However, the set of combinations of time zone name and language tag for which localized time zone names are available is implementation dependent.
+    </p>
+
+    <emu-clause id="sec-isvalidtimezonename">
+      <h1>IsValidTimeZoneName (timeZone)</h1>
+
+      <p>
+        The IsValidTimeZoneName abstract operation verifies that the _timeZone_ argument (which must be a String value) represents a valid Zone or Link name of the IANA Time Zone Database.
+      </p>
+      <p>
+        The abstract operation returns true if _timeZone_, converted to upper case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>, is equal to one of the Zone or Link names of the IANA Time Zone Database, converted to upper case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>. It returns false otherwise.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-canonicalizetimezonename">
+      <h1>CanonicalizeTimeZoneName</h1>
+
+      <p>
+        The CanonicalizeTimeZoneName abstract operation returns the canonical and case-regularized form of the _timeZone_ argument (which must be a String value that is a valid time zone name as verified by the IsValidTimeZoneName abstract operation). The following steps are taken:
+      </p>
+
+      <emu-alg aoid="canonicalizetimezonename-alg">
+        1. Let _ianaTimeZone_ be the Zone or Link name of the IANA Time Zone Database such that _timeZone_, converted to  upper case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>, is equal to _ianaTimeZone_, converted to upper case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
+        2. If _ianaTimeZone_ is a Link name, then
+          1. let _ianaTimeZone_ be the corresponding Zone name as specified in the "backward" file of the IANA Time Zone Database.
+        3. If _ianaTimeZone_ is *"Etc/UTC"* or *"Etc/GMT"*, then
+          1. return *"UTC"*.
```

Comment:
- Merge this back with step 3. 
- "Etc/UTC", "Etc/GMT" and "UTC" should be code/monospaced font